File: QA-project/backend/app/models/analytics_results.py
from sqlalchemy import Column, Integer, Float, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from app.database import Base


class AnalyticsResult(Base):
    __tablename__ = "analytics_results"

    result_id = Column(Integer, primary_key=True, autoincrement=True)
    dataset_id = Column(Integer, ForeignKey("datasets.dataset_id", ondelete="CASCADE"), nullable=False)
    reopen_rate = Column(Float)
    avg_resolution_time = Column(Float)
    defect_leakage_rate = Column(Float)
    computed_at = Column(DateTime, server_default=func.now(), nullable=False)

    # Relationships
    dataset = relationship("Dataset", back_populates="analytics_results")

File: QA-project/backend/src/db/schema/analytics-results.ts
import {
  pgTable,
  serial,
  integer,
  real,
  timestamp,
} from "drizzle-orm/pg-core";
import { relations } from "drizzle-orm";
import { datasets } from "./datasets";

export const analyticsResults = pgTable("analytics_results", {
  resultId: serial("result_id").primaryKey(),
  datasetId: integer("dataset_id")
    .notNull()
    .references(() => datasets.datasetId, { onDelete: "cascade" }),
  reopenRate: real("reopen_rate"),
  avgResolutionTime: real("avg_resolution_time"),
  defectLeakageRate: real("defect_leakage_rate"),
  computedAt: timestamp("computed_at").defaultNow().notNull(),
});

export const analyticsResultsRelations = relations(analyticsResults, ({ one }) => ({
  dataset: one(datasets, {
    fields: [analyticsResults.datasetId],
    references: [datasets.datasetId],
  }),
}));
